### DISCUSSION TITLE
Alias in nuxt Config doesn't warn you when using nuxt reserved aliases and causes lots of cryptic errors.

### DISCUSSION THREAD

**Discussion by rmannjbs** [nuxt/nuxt#29031] [Ideas]
By happen chance today I happen to create an alias in my nuxt config for "@nuxt" pointing to a shared folder named "nuxt" we use for sharing modules/plugins with other apps.  And then I noticed later that the site wouldn't run locally anymore and I just got a cryptic error like this

"unable to load devalue from ../nuxt/devalue"

And I had no idea why, I chased this for a long time trying to figure it out and then I realized that the /nuxt was my folder not 'nuxts' folder and that @nuxt was a reserved nuxt alias that it uses itself, so nitro was then trying to load all kinds of stuff out of my nuxt folder and is why it completely broke everything.

So the discussion here isn't me complaining that I can't use @nuxt, it's that we need better warnings.

Ideally I think that if you use a reserved alias in your alias config it should crash right there and descriptively tell me "You can't use reserved aliases for aliases, and then list all the reserved aliases."

I think it would be a nice addition.

Currently nuxt will let you create a reserved alias and point it somewhere else and basically allows you to hijack the entire core of how nuxt/nitro works.  Like you can point nitro at a different runtime folder etc etc..

Maybe that should continue to work, it's pretty powerful, but a warning would be really nice.

**Comment by rmannjbs** [comment]
Edit: Tired... So this is really about aliases that basically are overriding packages in node modules like "@nuxt" in this example.

I think it would be really cool if "nuxi prepare" could warn you about this, spit out a table like

[Aliases Warnings]
| Alias       | Warning: Overrides Node Package "Alias"            |
|-------------|----------------------------------------------------|
| @nuxt       | Warning: Overrides Node Package "@nuxt"            |
| scroll      | Warning: Overrides Node Package "scroll"           |
| lodash      | Warning: Overrides Node Package "lodash"           |
| axios       | Warning: Overrides Node Package "axios"            |
| moment      | Warning: Overrides Node Package "moment"           |
| vue         | Warning: Overrides Node Package "vue"              |
| vue-router  | Warning: Overrides Node Package "vue-router"       |
| vuex        | Warning: Overrides Node Package "vuex"             |
| express     | Warning: Overrides Node Package "express"          |
| react       | Warning: Overrides Node Package "react"            |
| react-dom  | Warning: Overrides Node Package "react-dom"         |
| mongoose    | Warning: Overrides Node Package "mongoose"         |

**Reply by TheAlexLichter** [reply]
Sounds like a reasonable DX improvement! Mind raising an issue? ☺️
